### PR TITLE
fix: correct model_id in chat

### LINKED ISSUE #1372: bug: No engine field in request body
### Problem Statement

I'm sending a chat/completions request to the cortex.cpp endpoints, setting up my daily use tools. The request is fully OpenAI compatible. I always receive an error response similar to this:

```json
{
    "message": "No engine field in request body"
}
```

### Feature Idea

Since the cortex.cpp manages to retrieve model information, it should select the engine accordingly. This eliminates the need for an extra `engine` field in the request, ensuring that 100% OpenAI-compatible products can utilize the endpoint.

### PR DESCRIPTION
**Pull request by vansangpfiev**
## Describe Your Changes

- Add `/models/start` and `/models/stop` endpoint
- Fallback to cortex.llamacpp if engine is not in request body
- Use model_handle for model chat id

## Fixes Issues

- fixes https://github.com/janhq/cortex.cpp/issues/1372

## Self Checklist

- [ ] Added relevant comments, esp in complex areas
- [ ] Updated docs (for bug fixes / features)
- [ ] Created issues for follow-up changes or refactoring needed

### CODE REVIEW

File: engine/commands/chat_completion_cmd.h [hunk 1]
```diff
@@ -9,8 +9,8 @@ class ChatCompletionCmd {
  public:
   void Exec(const std::string& host, int port, const std::string& model_handle,
             std::string msg);
-  void Exec(const std::string& host, int port, const config::ModelConfig& mc,
-            std::string msg);
+  void Exec(const std::string& host, int port, const std::string& model_handle,
+            const config::ModelConfig& mc, std::string msg);
```
Comment (namchuai) [inline engine/commands/chat_completion_cmd.h:13]: should be `const std::string& msg` ?

Comment (vansangpfiev) [reply engine/commands/chat_completion_cmd.h:13]: we use std::move, so it is not a problem. But I think should use `const std::string&` like other parameters. Will fix it shortly

**Review by namchuai — APPROVED** [review]
✅ lgtm

File: engine/commands/chat_completion_cmd.h (hunk 1 above)
Comment (vansangpfiev) [reply engine/commands/chat_completion_cmd.h:13]: ah no, it is intended to check the empty condition here
```
std::string user_input = std::move(msg);
      if (user_input.empty()) {
```

**Pull request merged**